GitUI/UserControls/BlameControl.cs

```diff
@@ -388,6 +434,44 @@ private void BuildAuthorLine(GitBlameLine line, StringBuilder lineBuilder, strin
             }
         }
 
+        private IList<Color> GetGradients()
+        {
+            // Color chosen from: https://colorbrewer2.org/#type=sequential&scheme=Greens&n=7
+            return new[]
+            {
+                Color.FromArgb(247, 252, 245),
+                Color.FromArgb(199, 233, 192),
+                Color.FromArgb(161, 217, 155),
+                Color.FromArgb(116, 196, 118),
+                Color.FromArgb(65, 171, 93),
+                Color.FromArgb(35, 139, 69),
+                Color.FromArgb(0, 68, 27),
+            }.Select(ColorHelper.AdaptBackColor).ToList();
+        }
+
+        public DateTime ArtificialOldBoundary => DateTime.Now.AddYears(-3);
+
+        private List<GitBlameEntry> CalculateBlameGutterData(IReadOnlyList<GitBlameLine> blameLines)
+        {
+            var mostRecentDate = DateTime.Now.Ticks;
+            var artificialOldBoundary = ArtificialOldBoundary;
+            var gitBlameDisplays = new List<GitBlameEntry>(blameLines.Count);
+            var colors = GetGradients();
```

Comment:
Why is this called multiple times? This array must be static and initialised in a static ctor (could be lazy).